integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -312,16 +312,20 @@ def _get_response(self, response_body: GenerateContentResponse) -> List[ChatMess
         :returns: The extracted responses.
         """
         replies = []
+        metadata = response_body.to_dict()
+        [candidate.pop("content", None) for candidate in metadata["candidates"]]
         for candidate in response_body.candidates:
             for part in candidate.content.parts:
                 if part.text != "":
-                    replies.append(ChatMessage.from_assistant(part.text))
+                    replies.append(ChatMessage(content=part.text, role=ChatRole.ASSISTANT, name=None, meta=metadata))
                 elif part.function_call is not None:
+                    metadata["function_call"] = part.function_call
                     replies.append(
                         ChatMessage(
                             content=dict(part.function_call.args.items()),
                             role=ChatRole.ASSISTANT,
                             name=part.function_call.name,
```

Comment:
It would be nice if we can avoid using name field. It was used early at the beginning for function calling, but ideally we should remove it soon.

Reply:
At the moment all ChatGenerators have this format for responses. I think for consistency we can keep `name` as long as its not removed from everywhere. Will be less confusing this way?